src/V3SplitVar.cpp

```diff
@@ -0,0 +1,719 @@
+        UINFO(3, "Start analyzing module " << nodep->prettyName() << '\n');
+        std::vector<std::pair<AstPragma*, AstVar*> > vars = ScanPragmaVisitor::scan(nodep);
+        for (size_t i = 0; i < vars.size(); ++i) {
+            AstPragma* const pragmap = vars[i].first;
+            AstVar* const varp = vars[i].second;
+            if (pragmap->pragType() != AstPragmaType::SPLIT_VAR) continue;  // nothing to do
+            UASSERT_OBJ(varp, pragmap,
+                        "Stray pragma must have been consumed in SplitUnpackedVarVisitor");
+            if (!canSplit(varp, true)) {
+                pragmap->v3warn(SPLITVAR,
+                                "Pragma split_var is specified on a variable whose type is "
+                                "unsupported or public. "
+                                "Packed portion must be an aggregate type of bit or logic.");
+            } else {  // finally find a good candidate
+                UINFO(3, varp->prettyNameQ() << " is added to candidate list.\n");
+                m_refs.insert(std::make_pair(varp, PackedVarRef(varp)));
+            }
+            pragmap->unlinkFrBack()->deleteTree();  // consume the pragma here anyway.
+            VL_DANGLING(vars[i].first);
+        }
+        iterateChildren(nodep);
+        split();
+        m_modp = NULL;
+    }
+    virtual void visit(AstVarRef* nodep) VL_OVERRIDE {
+        AstVar* const varp = nodep->varp();
+        vl_unordered_map<AstVar*, PackedVarRef>::iterator refit = m_refs.find(varp);
+        if (refit == m_refs.end()) return;  // variable without split_var pragma
+        UASSERT_OBJ(nodep->lvalue() == m_isLhs, nodep,
+                    (m_isLhs ? 'l' : 'r') << "value is expected");
+        const AstBasicDType* const basicp = refit->second.basicp();
+        refit->second.append(PackedVarRef::Entry(nodep, basicp->lsb(), basicp->width()), m_isLhs);
+    }
+    virtual void visit(AstSel* nodep) VL_OVERRIDE {
+        AstVarRef* const vrefp = VN_CAST(nodep->fromp(), VarRef);
+        if (!vrefp) return;
+
+        AstVar* const varp = vrefp->varp();
+        vl_unordered_map<AstVar*, PackedVarRef>::iterator refit = m_refs.find(varp);
+        if (refit == m_refs.end()) return;  // variable without split_var pragma
+
+        AstConst* const consts[2] = {constifyIfNot(nodep->lsbp()), constifyIfNot(nodep->widthp())};
+
+        if (consts[0] && consts[1]) {  // OK
+            refit->second.append(
+                PackedVarRef::Entry(nodep, consts[0]->toSInt() + refit->second.basicp()->lsb(),
+                                    consts[1]->toUInt()),
+                m_isLhs);
+        } else {
+            nodep->v3warn(SPLITVAR, "Variable "
+                                        << vrefp->prettyNameQ()
+                                        << " will not be split"
+                                           " because bit range cannot be determined statically.");
+            m_refs.erase(varp);
+        }
+    }
```

Comment:
As noted should never happen, can assert

Reply:
When split_var is specified on a memory model, the accessed index is not constatnt. 
So this may happen and tested in t_split_var_1_bad.
> logic [1023:0] [7:0] memory;  /* verilator split_var */
> always @(posedge clk) begin
>    if (we) memory[waddr] <= wd;
>    rd <= memory[raddr];
> end